runtime/src/bank.rs

```diff
@@ -2764,6 +2765,32 @@ impl Bank {
         }
     }
 
+    #[allow(dead_code)]
+    fn hash_rewards_into_partitions(
+        stake_rewards: &StakeRewards,
```

Comment:
do we need to retain `stake_rewards` after this operation?  seems like we can just consume them here.

Reply:
without ref, it will still do a move instead of copy.
I have benchmarked this. The performance is similar, with moving being slightly slower than copy.

Reply:
i'm less concerned about this call site than the unnecessary cloning taking a ref here causes later.

Reply:
I am talking about the same thing. 
In the following for loop, cloning is actually faster than moving the element from the original vec.

Reply:
I am surprised too. 

```
move: 45957us
copy: 45955us
```

Reply:
heh, that's buried pretty deep in the sampling noise

Reply:
This method should indeed  take a value not a reference.
That 2us is not significant.

Reply:
`stake_rewards: StakeRewards` now.